src/components/page/page_header/page_header.test.tsx

```diff
@@ -124,4 +124,28 @@ describe('OuiPageHeader', () => {
       });
     });
   });
+
+  it('should console deprecation warning', () => {
+    console.warn = jest.fn();
+
+    mount(
+      <OuiPageHeader iconType="dashboardApp" iconProps={{ color: 'red' }} />
+    );
+
+    expect(console.warn).toHaveBeenCalledTimes(2);
+    expect(console.warn).toHaveBeenCalledWith(
+      '[DEPRECATED] The `iconType` prop is deprecated and will be removed in v2.0.0'
+    );
+    expect(console.warn).toHaveBeenCalledWith(
+      '[DEPRECATED] The `iconProps` prop is deprecated and will be removed in v2.0.0'
+    );
+  });
+
+  it('should not console deprecation warning', () => {
+    console.warn = jest.fn();
+
+    mount(<OuiPageHeader />);
+
+    expect(console.warn).not.toHaveBeenCalled();
+  });
```

Comment:
Maybe we could have tests for all permutations?

I.e.
 - `iconType` and `iconProps`
 - only `iconType`
 - neither